docker-images/oncoanalyser/assets/run.sh

```diff
@@ -248,17 +356,88 @@ region = ap-southeast-2
 no_check_bucket = true
 EOF
 
-  src_fp=$(jq -r '.bucketName + "/" + .keyPrefix + "'${gds_fp##*/}'"' <<< ${creds})
+  src_fp="$( \
+    jq --raw-output \
+      --arg gds_fp "${gds_fp##*/}" \
+      '\(.bucketName)/\(.keyPrefix)\($gds_fp)' \
+      <<< "${creds}" \
+  )"
 
   if [[ ${gds_fp} =~ .*bam$ ]]; then
     echo "staging ${gds_fp}.bai to s3://${dst_fp}.bai" 1>&2
-    rclone copy --s3-upload-concurrency 8 ica:${src_fp}.bai aws:${dst_dp}/
+    rclone copy --s3-upload-concurrency 8 \
+      "ica:${src_fp}.bai" "aws:${dst_dp}/"
   fi
 
   echo "staging ${gds_fp} to s3://${dst_fp}" 1>&2
-  rclone copy --s3-upload-concurrency 8 ica:${src_fp} aws:${dst_dp}/
+  rclone copy --s3-upload-concurrency 8 \
+    "ica:${src_fp}" "aws:${dst_dp}/"
+}
+
+samplesheet_wgs_entries() {
+  echo "${subject_id}_${tumor_wgs_id},${subject_id},${tumor_wgs_id},tumor,wgs,bam,${input_fps['tumor_wgs_bam']}"
+  echo "${subject_id}_${tumor_wgs_id},${subject_id},${normal_wgs_id},normal,wgs,bam,${input_fps['normal_wgs_bam']}"
+}
+
+samplesheet_wts_entries() {
+  echo "${subject_id}_${1},${subject_id},${tumor_wts_id},tumor,wts,fastq_fwd,${input_fps['tumor_wts_fastq_fwd']}"
+  echo "${subject_id}_${1},${subject_id},${tumor_wts_id},tumor,wts,fastq_rev,${input_fps['tumor_wts_fastq_rev']}"
+}
+
+# Final upload data function
+upload_data() {
+  aws s3 sync \
+    --no-progress \
+    --no-follow-symlinks \
+    --exclude='software/*' \
+    --exclude='assets/*' \
+    --exclude='work/*' \
+    ./ "${output_dir}/"
 }
 
+## END FUNCTIONS ##
+
+## LOCAL EXECUTOR WORKAROUND ##
+
+# When Nextflow runs a job using the local executor with Docker enabled, I have configured behaviour such that that a
+# new Docker container from the host service is launched. This means that all local Nextflow processes inherit the EC2
+# instance IAM profile, which can only be set prior at the Batch compute environment creation; in this case that is the
+# non-permissive Nextflow pipeline role. This means to run Nextflow processes locally that can r/w to S3 (e.g. when
+# using Fusion, S3 output directory, etc), we must set the EC2 instance IAM role to a profile with such permissions.
+# Here I associate the instance with the OncoanalyserStack task role. There may be better approaches to achieve this.
+
+# NOTE(SW): this is to be updated manually for now until build process is placed into CodePipeline
```

Comment:
Can remove this comment now 😃

Reply:
Done in 7acc9c13da2d7887fd0065cbb5f7971d1f22d4dc